### ISSUE TITLE
Telemetry: Agent configuration - Update the documentation

### ISSUE THREAD

**Issue by o0Ignition0o** [apollographql/router#4177]
Followup to #3950, the tracing configuration documentation is out of date:

[trace_config](https://www.apollographql.com/docs/router/configuration/tracing/#trace-config) doesn't exist anymore since #4044 

![image](https://github.com/apollographql/router/assets/9465678/dd184ae7-dd12-4876-b1a8-baeea92f5a94)

This image claims jaeger: agent: endpoint takes a socket address, yet shows a hostname+port, which the router won't allow.

As a side effect of us using serde untagged enums, errors raised while trying to deserialize to a variant greet the user with a difficult to diagnose error message:

```
plugin apollo.telemetry could not be configured: data did not match any variant of untagged enum Config
```

**Comment by oliverjumpertz** [comment]
This has actually already been solved by with #3962 :)

Just checked when I wanted to quickly fix the docs.

**Comment by abernix** [comment]
Thanks for pointing that out, @oliverjumpertz!  Closing as it's already been fixed!  (Thanks, @BrynCooke , too. ;) )
